src/ImageIO/CGImageAnimation.cs

```diff
@@ -0,0 +1,128 @@
+        [Introduced (PlatformName.iOS, 13, 0, PlatformArchitecture.All)]
+        [BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
+        public virtual int CGAnimateImage (NSUrl url, NSDictionary options, [BlockProxy (typeof (NIDCGImageSourceAnimationBlock))] CGImageSourceAnimationBlock block) {
+            if (url == null)
+                throw new ArgumentNullException ("url");
+            if (block == null)
+                throw new ArgumentNullException ("block");
+
+            int ret;
+            unsafe {
+                BlockLiteral * block_ptr_block;
+                BlockLiteral block_block;
+                block_block = new BlockLiteral ();
+                block_ptr_block = & block_block;
+                block_block.SetupBlockUnsafe (SDCGImageSourceAnimationBlock.Handler, block);
+
+                try {
+                    ret = CGAnimateImageAtURLWithBlock (url.Handle, options == null ? IntPtr.Zero : options.Handle, (IntPtr) block_ptr_block);
+                } finally {
+                    block_ptr_block -> CleanupBlock ();
+                }
+            }
+
+            return ret;
+        }
+
+        [Introduced (PlatformName.MacOSX, 10, 15, PlatformArchitecture.All)]
+        [Introduced (PlatformName.iOS, 13, 0, PlatformArchitecture.All)]
+        [BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
+        public virtual int CGAnimateImage (NSData data, NSDictionary options, [BlockProxy (typeof (NIDCGImageSourceAnimationBlock))] CGImageSourceAnimationBlock block) {
+            if (data == null)
+                throw new ArgumentNullException ("data");
+            if (block == null)
+                throw new ArgumentNullException ("block");
+
+            int ret;
+            unsafe {
+                BlockLiteral * block_ptr_block;
+                BlockLiteral block_block;
+                block_block = new BlockLiteral ();
+                block_ptr_block = & block_block;
+                block_block.SetupBlockUnsafe (SDCGImageSourceAnimationBlock.Handler, block);
+
+                try {
+                    ret = CGAnimateImageDataWithBlock(data.Handle, options == null ? IntPtr.Zero : options.Handle, (IntPtr) block_ptr_block);
+                } finally {
+                    block_ptr_block -> CleanupBlock ();
+                }
+            }
+ 
+            return ret;
+        }
+
+        //
+        // This class bridges native block invocations that call into C#
+        //
+        static internal class SDCGImageSourceAnimationBlock {
+            static internal readonly DCGImageSourceAnimationBlock Handler = Invoke;
+
+            [MonoPInvokeCallback (typeof (DCGImageSourceAnimationBlock))]
+            static unsafe void Invoke (IntPtr block, nint arg0, IntPtr arg1, [System.Runtime.InteropServices.MarshalAs (System.Runtime.InteropServices.UnmanagedType.I1)] out bool done) {
```

Comment:
Lets not use the generator variable names, I'm sure we know what arg0 and arg1 are. And using meaningful names will make our future lifes better.